### DISCUSSION TITLE
Using Lightning DataModules

### DISCUSSION THREAD

**Discussion by simon-forb** [pyg-team/pytorch_geometric#9235] [Q&A]
I have the following minimal code example:

```python
from lightning import LightningDataModule
from torch.utils.data import random_split
from torch_geometric.data.lightning import LightningDataset
from torch_geometric.datasets import TUDataset
import lightning as L

trainer = L.Trainer()
reddit_dataset = TUDataset("./data", "REDDIT-BINARY")
train_set, val_set, test_set = random_split(reddit_dataset, [.7, .1, .2])
dm = LightningDataset(train_set, val_set, test_set, batch_size=32)

trainer.fit(model, datamodule=dm)
```

Which gives me the following error:

```
TypeError: An invalid dataloader was passed to `Trainer.fit(train_dataloaders=...)`. Found LightningDataset(train_dataset=<torch.utils.data.dataset.Subset object at 0x773852bfe710>, val_dataset=<torch.utils.data.dataset.Subset object at 0x773852c35110>, test_dataset=<torch.utils.data.dataset.Subset object at 0x773852bffa50>, batch_size=32, num_workers=0, pin_memory=True, persistent_workers=False).
```

I would have expected this is how `LightningDataset` is supposed to be applied.

My code works when I create standard `DataLoader`s from PyG.

**Version:**  pyg 2.5.2 py311_torch_2.2.0_cu121

**Comment by rusty1s** [comment]
Which PL version are you using?

**Reply by simon-forb** [reply]
This is a very recent environment, it uses

pytorch-lightning=2.2.2=pyhd8ed1ab_0

**Comment by brenerrr** [comment]
I am having the same issue. 

torch: 2.1.1 
lightning: 2.4.0 
torch_geometric: 2.4.0

**Reply by akihironitta** [reply]
Mind updating it to `torch_geometric==2.6.0` and retrying it? I cannot reproduce this with the PyTorch Lightning version.

**Comment by zyjOwO** [comment]
I reimplemented similar issue while using Ray and PyG.
[Reason]
Ray (or your code) imports the [lightning] package, but PyG imports the [pytorch_lightning] package.
[Solution]
pyg.data.lightning.datamodule.py (line14) 
from pytorch_lightning import LightningDataModule as PLLightningDataModule
==> from lightning import LightningDataModule as PLLightningDataModule
[Future]
[pytorch_lightning] got renamed to [lightning]. The newer PyG (>2.7.0) could consider it for avoiding such conflicts. @rusty1s

**Reply by akihironitta** [reply]
@zyjOwO Thanks for reporting this issue. Since `lightning` used to contain totally unnecessary code/dependencies until https://github.com/Lightning-AI/pytorch-lightning/pull/20039, I'd prefer using `pytorch-lightning`, which the Lightning team has promised to keep supporting, but we could potentially consider adopting your suggested change. Feel free to send a PR to support both `pytorch_lightning` and `lightning` import names if you're interested in contributing to PyG!

**Reply by zyjOwO** [reply]
Thanks for your prompt response. I'd prefer using [lightning]. The Lightning team won't show [pytorch_lightning] anymore in new docs (lightning>=2.0.0). The Ray team used the following solution to support both packages. But it cannot avoid such conflicts totally.
[Solution]
try:
    from lightning import *
except ModuleNotFoundError:
    from pytorch_lightning import *
